python/cugraph/cugraph/gnn/data_loading/dist_sampler.py

```diff
@@ -0,0 +1,577 @@
+
+        w.wait()
+        if isinstance(label_list, list):
+            label_list = torch.concat(label_list)
+        return label_list, label_to_output_comm_rank
+
+    def get_start_batch_offset(
+        self, local_num_batches: int, assume_equal_input_size: bool = False
+    ) -> Tuple[int, bool]:
+        """
+        Gets the starting batch offset to ensure each rank's set of batch ids is
+        disjoint.
+
+        Parameters
+        ----------
+        local_num_batches: int
+            The number of batches for this rank.
+        assume_equal_input_size: bool
+            Whether to assume all ranks have the same number of batches.
+
+        Returns
+        -------
+        Tuple[int, bool]
+            The starting batch offset (int)
+            and whether the input sizes on each rank are equal (bool).
+
+        """
+        input_size_is_equal = True
+        if self.is_multi_gpu:
+            rank = torch.distributed.get_rank()
+            world_size = torch.distributed.get_world_size()
+
+            if assume_equal_input_size:
+                t = torch.full(
+                    (world_size,), local_num_batches, dtype=torch.int64, device="cuda"
+                )
+            else:
+                t = torch.empty((world_size,), dtype=torch.int64, device="cuda")
+                local_size = torch.tensor(
+                    [local_num_batches], dtype=torch.int64, device="cuda"
+                )
+
+                torch.distributed.all_gather_into_tensor(t, local_size)
+                if (t != local_size).any():
+                    input_size_is_equal = False
+                    if rank == 0:
+                        warnings.warn(
+                            "Not all ranks received the same number of batches. "
+                            "This might cause your training loop to hang "
+                            "due to uneven inputs."
+                        )
+
+            return (0 if rank == 0 else t.cumsum(dim=0)[rank - 1], input_size_is_equal)
+        else:
+            return 0, input_size_is_equal
+
+    def sample_from_nodes(
+        self,
+        nodes: TensorType,
+        *,
+        batch_size: int = 16,
+        random_state: int = 62,
+        assume_equal_input_size: bool = False,
+    ):
+        """
+        Performs node-based sampling.  Accepts a list of seed nodes, and batch size.
+        Splits the seed list into batches, then divides the batches into call groups
+        based on the number of seeds per call this sampler was set to use.
+        Then calls sample_batches for each call group and writes the result using
+        the writer associated with this sampler.
+
+        Parameters
+        ----------
+        nodes: TensorType
+            Input seeds (node ids).
+        batch_size: int
+            The size of each batch.
+        random_state: int
+            The random seed to use for sampling.
+        """
+        nodes = torch.as_tensor(nodes, device="cuda")
```

Comment:
This is also causing problems in test environments that don't have `torch`:
```
/opt/conda/envs/test/lib/python3.9/site-packages/cugraph/gnn/data_loading/dist_sampler.py:396: in sample_from_nodes
    nodes = torch.as_tensor(nodes, device="cuda")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <cugraph.utilities.utils.MissingModule object at 0x7fed4e169d90>
attr = 'as_tensor'

    def __getattr__(self, attr):
>       raise RuntimeError(f"This feature requires the {self.name} " "package/module")
E       RuntimeError: This feature requires the torch package/module

/opt/conda/envs/test/lib/python3.9/site-packages/cugraph/utilities/utils.py:410: RuntimeError
```
I don't have a good suggestion other than to skip all tests that use this method.  Looks like `torch.split()`, `torch.distributed.all_reduce()` and maybe others are also used and would cause problems.